api/src/transactions.rs

```diff
@@ -216,6 +292,45 @@ impl Transactions {
         }
     }
 
+    pub async fn simulate(self, txn: SignedTransaction) -> Result<impl Reply, Error> {
+        let state_view = &*self.context.move_resolver()?;
+        let user_txn = aptos_types::transaction::Transaction::UserTransaction(txn.clone());
+        let res = AptosVM::execute_block_and_keep_vm_status(vec![user_txn.clone()], state_view);
```

Comment:
I think what we'll want to add is a special function in the adapter that basically unifies a few bits of code:

* Make a fake Transaction(SignedTransaction) from a RawTransaction
* entry point is `AptosVM::simulate_transaction(txn: Transaction, ...)`
* convert the txn to a PreprocessedTransaction
* call into `AptosVM::execute_single_transaction`
* return all the way out

Reply:
yeah `AptosVM::simulate_transaction(txn: Transaction, ...)` is definitely worth it. Adding now.